### ISSUE TITLE
Killing the current leader doesn't allow the network to continue

### ISSUE THREAD

**Issue by sakridge** [solana-labs/solana#2383]
#### Problem

Start a network with 2 or more nodes, then ctrl-c the leader node. The network will have trouble starting or finding a new leader.

**Comment by sagar-solana** [comment]
@sakridge this is blocked on the Forking stuff being implemented _and_ validators running poh recorders so that they can figure out when a leader has "timed-out".

**Comment by sagar-solana** [comment]
Duplicate of #2570 

Closing this one since #2570 has a proposed solution.
